lib/SIL/SILVerifier.cpp

```diff
@@ -4345,6 +4345,40 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
     
     SILInstructionVisitor::visitSILBasicBlock(BB);
+
+    // Visit the instructions in the BB, to find holes in the lexical scopes.
+    // We use a fairly simple heuristic here (Explain heuristic).
+    llvm::DenseSet<const SILDebugScope *> AlreadySeenScopes;
+    if (BB->getParent()->getEffectiveOptimizationMode() !=
+        OptimizationMode::NoOptimization)
+      return;
+    if (BB->empty())
+      return;
+    SILInstruction &First = *BB->begin();
+    const SILDebugScope *LastSeenScope = First.getDebugScope();
+    assert(LastSeenScope && "Invalid top-level debug scope");
+    AlreadySeenScopes.insert(LastSeenScope);
+    for (SILInstruction &SI : *BB) {
+      auto *DS = SI.getDebugScope();
+      assert(DS && "Each instruction should have a debug scope");
+      if (!AlreadySeenScopes.count(DS)) {
+        AlreadySeenScopes.insert(DS);
+        LastSeenScope = DS;
+        continue;
+      } else {
+        PointerUnion<const SILDebugScope *, SILFunction *> Parent = LastSeenScope->getParent();
+        if (auto *ParentScope = Parent.dyn_cast<const SILDebugScope *>()) {
+          if (ParentScope == DS) {
+            LastSeenScope = DS;
+            continue;
+          }
+        }
+      }
+      if (DS != LastSeenScope) {
+        llvm::dbgs() << "Broken instruction!\n"; SI.dump();
+        require(DS == LastSeenScope, "Found an hole in debug lexical scopes");
```

Comment:
`require` is printing the wrong instruction (always the last one in the block). We probably need an overload where we pass the instruction to print.